lib/vmdb/settings/database_source.rb

```diff
@@ -3,23 +3,60 @@ class Settings
     class DatabaseSource
       include Vmdb::Logging
 
-      attr_reader :resource
+      attr_reader :resource_instance, :settings_holder_class
 
-      def initialize(resource)
-        @resource = resource
+      # @param resource [MiqRegion, Zone, MiqServer] the resource
+      # @param klass [Symbol] class name of object which provides access to SettingsChange for resource
+      #   possible values of klass are {DatabaseSource::SETTINGS_HIERARCHY the setting hierarchy}
+      #
+      #   Example:
+      #   1. resource is instance of MiqServer and klass is 'Zone', than settings will be loaded
+      #      from resource.zone.settings_changes
+      #   2. resource is instance of MiqServer and klass is 'MiqRegion' than settings will be loaded
+      #      from resource.miq_region.settings_changes
+      def initialize(resource, klass)
+        @resource_instance = resource
+        @settings_holder_class = klass
```

Comment:
As discussed, let's rename this ivar to `settings_holder_class_name`, the param name to `class_name`, and also call .to_sym here to ensure it's the right value.  You could also verify the value and raise ArgumentError if you wanted to, but it's not really necessary.  Also, don't forget the change the documentation in 4 places when you rename `klass` to `class_name`